extensions/amp-experiment/1.0/amp-experiment.js

```diff
@@ -48,21 +50,43 @@ export class AmpExperiment extends AMP.BaseElement {
       const variantsService = responses[0];
       const enabled = responses[1];
 
-      if (!enabled) {
-        user().error(TAG, 'Experiment amp-experiment-1.0 is not enabled.');
+      let config = dict({});
 
-        // Ensure downstream consumers don't wait for the promise forever.
-        variantsService.init(Promise.resolve({}));
+      try {
+        config = this.getConfig_();
 
-        return Promise.reject('Experiment amp-experiment-1.0 is not enabled.');
-      }
+        if (!enabled) {
+          user().error(TAG, 'Experiment amp-experiment-1.0 is not enabled.');
+
+          // Ensure downstream consumers don't wait for the promise forever.
+          variantsService.init(
+            Promise.resolve(this.getEmptyExperimentToVariant_(config))
+          );
+
+          return Promise.reject(
+            'Experiment amp-experiment-1.0 is not enabled.'
+          );
+        }
+
+        const ampdoc = this.getAmpDoc();
+
+        // All experiments can be disabled by a hash param
+        const viewer = Services.viewerForDoc(ampdoc);
+        const override = viewer.getParam(
+          ATTR_PREFIX + '_disable_all_experiments_'
+        );
+        if (override !== undefined) {
+          variantsService.init(
+            Promise.resolve(this.getEmptyExperimentToVariant_(config))
+          );
+          return;
+        }
 
-      try {
-        const config = this.getConfig_();
         const experimentToVariant = Object.create(null);
         const variants = Object.keys(config).map(experimentName => {
           return allocateVariant(
-            this.getAmpDoc(),
+            ampdoc,
+            viewer,
```

Comment:
nit: Do we really need to pass viewer to the function? We could easily get the service from ampdoc.

Reply:
So I decided to pass it in, for two reasons:

1. We need access to viewer anyways now to check for `_disable_all_experiments_` on the URL. And I thought it would be better practice to pass it in to remove dependencies from our help 'variants.ts' file.

2.  We want to be sensitive about performance. Trying to re-grab the viewer every single time would be slower than just passing it in. 

Totally open to doing it inside of `allocateVariant` though. It is up to you 😄 But personally, I think this is the way to go for now unless you have strong feelings about it.

Reply:
I don't have a strong feeling about this. So feel free to merge. 
But I'd be curious to know the performance difference between passing the viewer versus getting the viewer from ampdoc. 
Personally I imagine `Services.viewerForDoc(ampdoc)` is very cheap as it's only reading some stored value from doc. 

@choumx Could you shed us some light on the performance difference between the two. 1. passing as an argument, 2. retrieving using ampdoc.

Reply:
@zhouyx 

Asked @kristoferbaxter on this since they are also with me in Berlin 😄 

But they stated performance wise the change would be negligible, but mentioned they would personally hoist it out of the loop haha 😂 

I think we all don't have strong feelings on this, so, I'll go ahead and merge, and we can resolve if needed back in California 😄